### PR TITLE
I-#365-will enforce donation time within 24 hours when urgent selected

### LINKED ISSUE #365: Enforce 24-Hour Request Time for Urgent Requests
When a user selects "urgent" for a request, ensure the request time is within the next 24 hours. Prevent submission if the specified time exceeds this limit and provide a clear error message.  

**Tasks**:  
- [x] Add validation to check if the request time is within 24 hours when "urgent" is selected.  
- [x] Display an error message if the validation fails.  
- [x] Update API and UI logic to enforce this rule.  
- [x] Test the validation across different time zones and edge cases.  

**Acceptance Criteria**:  
- [ ] Users cannot submit an "urgent" request with a time beyond 24 hours.  
- [ ] Clear error messages are displayed when validation fails.  
- [ ] Tests confirm validation works correctly across all scenarios.

### PR DESCRIPTION
**Pull request by Rezwanul-Haque**
Fixes #365 

Resolves #
Used when the changes in the pull request indirectly resolve another issue.

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
When seeker creating post and select urgent post, we'll enforce the user to set the donation date within 24 hours.

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: clients/mobile/src/utility/validator.ts [hunk 1]
```diff
@@ -80,6 +81,29 @@ export const validateDonationDateTime = (donationDateTime: string): string | nul
   return null
 }
 
+export const validateDonationDateTimeWithin24Hours = (donationDateTime: string): string | null => {
+  const now = new Date()
+  const donationDate = new Date(donationDateTime)
+
+  const maxAllowedDate = new Date(now.getTime() + 24 * 60 * 60 * 1000)
```
Comment (mahinuzzaman) [inline clients/mobile/src/utility/validator.ts:88]: can we use 24 as a variable

**Review by mahinuzzaman — DISMISSED** [review]

**Comment by Rezwanul-Haque** [comment]
moving to this PR [https://github.com/CraftsmenLtd/BloodConnect/pull/396](#396)
